src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -218,14 +220,24 @@ def create_connectedk8s(cmd, client, resource_group_name, cluster_name, https_pr
     config_dp_endpoint = get_config_dp_endpoint(cmd, location)
 
     # Retrieving Helm chart OCI Artifact location
-    registry_path = os.getenv('HELMREGISTRY') if os.getenv('HELMREGISTRY') else utils.get_helm_registry(cmd, config_dp_endpoint, dp_endpoint_dogfood, release_train_dogfood)
+    if container_registry_repository:
+        if disable_auto_upgrade is False:
+            auto_upgrade_error = 'Disable auto upgrade when using custom repository'
+            telemetry.set_exception(exception=auto_upgrade_error, fault_type=consts.Custom_Repository_Disable_Auto_Upgrade_Fault_Type, summary=auto_upgrade_error)
+            raise ValidationError(auto_upgrade_error, recommendation='use --disable_auto_upgrade')
```

Comment:
I think the '[ArgumentUsageError](https://github.com/Azure/azure-cli/blob/9acf9d96bf3774bc2ebcae8043c48808cc7f68cf/src/azure-cli-core/azure/cli/core/azclierror.py#L157)' would make more sense here.  To look at all the supported error types in CLI, you may use [this](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md#error-type) link